pkg/clients/azure/azure.go

```diff
@@ -18,26 +18,38 @@ const (
 	CredentialsKeyTenantID       = "tenantId"
 	CredentialsKeyClientCert     = "clientCertificate"
 	CredentialsKeyClientCertPass = "clientCertificatePassword"
+
+	kubeloginCLIFlagServerID = "server-id"
 )
 
+func kubeloginTokenOptionsFromRESTConfig(rc *rest.Config) (*token.Options, error) {
+	opts := &token.Options{}
+
+	// opts are filled according to the provided args in the execProvider section of the kubeconfig
+	// we are parsing serverID from here
+	// add other flags if new login methods are introduced
+	fs := pflag.NewFlagSet("kubelogin", pflag.ContinueOnError)
```

Comment:
Looks like this complains while parsing parse kubelogin flags, due to undefined flags being present in the supplied arguments. I was able to overcome this with the following configuration on `pflag`
```suggestion
	fs := pflag.NewFlagSet("kubelogin", pflag.ContinueOnError)
	fs.ParseErrorsWhitelist = pflag.ParseErrorsWhitelist{UnknownFlags: true}
```

Of course, as I described in my [previous comment](https://github.com/crossplane-contrib/provider-helm/pull/207#discussion_r1447110675), ideally I believe it would be more robust if we could parse this through some `kubelogin`-maintained utility func and directly obtain all library related flags. Otherwise, we need to maintain the same list of `kubelogin` flags here.